src/lambda/logger.js

```diff
@@ -0,0 +1,110 @@
+import _ from 'lodash-firecloud';
+import aws from 'aws-sdk';
+
+import {
+  MinLog,
+  logToConsole,
+  serializeErr,
+  serializeTime
+} from 'minlog';
+
+let _makeLambdaSerializer = function({ctx}) {
+  return async function({entry}) {
+    entry.lambda = {
+      awsRequestId: ctx.awsRequestId,
+      functionName: ctx.functionName
+    };
+
+    return entry;
+  };
+};
+
+let _awsLoggerRE =
+  /^ *\[AWS ([^ ]+) ([^ ]+) ([^ ]+)s ([^ ]+) retries] ([^(]+)\(((?:.|\n)+)\)[^)]*$/;
+
+let _setupAwsLogger = function({ctx}) {
+  aws.config.logger = {
+    isTTY: false,
+    log: function(message) {
+      if (ctx.log.level() < ctx.log.levelToLevelCode('TRACE')) {
+        return;
+      }
+
+      let [
+        serviceIdentifier,
+        status,
+        delta,
+        retryCount,
+        operation,
+        params
+      ] = _awsLoggerRE.exec(message).slice(1);
+
+      try {
+        let paramsWithoutArrayLength = _.replace(params, /,\s+\[length\]:\s+\d+(\s+\])/g, '$1');
+        // eslint-disable-next-line no-eval
+        params = eval(`(${paramsWithoutArrayLength})`);
+      } catch (err) {
+        ctx.log.error({
+          err,
+          message,
+          params
+        });
+      }
+
+      ctx.log.trace('Making an AWS SDK call.', {
+        aws: {
+          serviceIdentifier,
+          status,
+          delta,
+          retryCount,
+          operation,
+          params
+        }
+      });
+    }
+  };
+};
+
+let _setupLongStacktraces = function({ctx}) {
+  if (ctx.log.level() >= ctx.log.levelToLevelCode('TRACE')) {
```

Comment:
Here and in other places, you expect `ctx.log.level()` to return a number, I guess, to compare it with level code for `TRACE`. But I can clearly see that the value it will return will be string like `INFO` (see `setup`, line 101).

Reply:
❤️